src/kakarot/library.cairo

```diff
@@ -101,6 +105,15 @@ namespace Kakarot {
 
         let env = Starknet.get_env(origin, gas_price);
 
+        let full_words_len = packed_data_len - 2;
+        let last_word_bytes = packed_data[packed_data_len - 2];
+        let last_word = packed_data[packed_data_len - 1];
```

Comment:
what is there is no need for a last word? you put 0 in the previous felt?

I feel this custom encoding a bit unsual (inserting a len in the middle of the data).

I think that it would be closed to a regular `bytes` encoding if we just put at `packed_data[0]` the `len` of the calldata (in bytes), then the packed data.
The final word is can just be shifted eventually with

```
(full_words, last_word_shift) = unsigned_div_rem(len, 31);
let last_word_shift = 32 - last_word_shift
```